images/virtualization-artifact/pkg/controller/vd/internal/attachee.go

```diff
@@ -52,8 +71,32 @@ func (h AttacheeHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (re
 		vd.Status.AttachedToVirtualMachines = append(vd.Status.AttachedToVirtualMachines, virtv2.AttachedVirtualMachine{
 			Name: vm.GetName(),
 		})
+
+		runningCondition, _ := service.GetCondition(vmcondition.TypeRunning.String(), vm.Status.Conditions)
+		if runningCondition.Status != metav1.ConditionFalse || vm.Status.Phase == virtv2.MachineStarting {
+			inUsed = true
+		} else {
+			kvvm, err := helper.FetchObject(ctx, types.NamespacedName{Namespace: vm.GetNamespace(), Name: vm.GetName()}, h.client, &virtv1.VirtualMachine{})
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("error getting kubevirt virtual machine: %w", err)
+			}
+
+			if kvvm.Status.StateChangeRequests != nil {
+				inUsed = true
+			}
+		}
+	}
+
+	if inUsed {
+		inUseCondition.Status = metav1.ConditionTrue
+		inUseCondition.Reason = vdcondition.InUseInRunningVirtualMachine
+	} else if inUseCondition.Reason == vdcondition.InUseInRunningVirtualMachine {
```

Comment:
Are you sure about this condition?

Reply:
Yes, because we dont want change condition, if this condition seted in anouther place.

Reply:
This code changed.